src/sentry/api/endpoints/project_autofix_create_codebase_index.py

```diff
@@ -0,0 +1,50 @@
+from __future__ import annotations
+
+import logging
+
+import requests
+from django.conf import settings
+from rest_framework.response import Response
+
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases.project import ProjectEndpoint
+from sentry.api.helpers.repos import get_repos_from_project_code_mappings
+from sentry.models.project import Project
+
+logger = logging.getLogger(__name__)
+
+from rest_framework.request import Request
+
+
+@region_silo_endpoint
+class ProjectAutofixCreateCodebaseIndexEndpoint(ProjectEndpoint):
+    publish_status = {
+        "POST": ApiPublishStatus.EXPERIMENTAL,
+    }
+    owner = ApiOwner.ML_AI
+    private = True
+
+    def post(self, request: Request, project: Project) -> Response:
+        """
+        Create a codebase index for for a project's repositories, uses the code mapping to determine which repositories to index
+        """
+        repos = get_repos_from_project_code_mappings(project)
+
+        for repo in repos:
```

Comment:
How long does each request take? I'm wondering if this might time out for large github orgs. Ours for example has nearly 400 repos

Reply:
The call to codebase index schedules a celery task to index it and isn't sync. Also because this uses the code mappings one call for this should only run for the project's repos that have code mappings, IIRC the sentry project only has `sentry` and `getsentry`

Reply:
Ah that's true, code mappings would be a much smaller number. Ideally we could send the entire list at once but this should work for now